### PR TITLE
Check attributes

### PR DESCRIPTION
**Pull request by m1n0**

### CODE REVIEW

File: soda/core/soda/common/attributes_handler.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+from soda.common.logs import Logs
+
+
+class AttributeHandler:
+    def __init__(self, logs: Logs) -> None:
+        self.logs = logs
+
+    def validate(self, attributes: dict[str, any], schema: list[dict]) -> tuple[dict[str, any]]:
+        "Validates given attributes against given schema. Returns a dict with valid and a dict with invalid attributes."
```
Comment (vijaykiran) [inline soda/core/soda/common/attributes_handler.py:13]: Did you mean this to be a docstring? In that case, please use """

File: soda/core/soda/common/attributes_handler.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    @staticmethod
+    def is_valid_datetime(value: any, schema: dict[str, any]) -> bool:
+        if isinstance(value, datetime):
+            return True
+
+        try:
+            datetime.fromisoformat(value)
+            return True
+        except Exception:
+            # Handle correct exception type and
```
Comment (vijaykiran) [inline soda/core/soda/common/attributes_handler.py:61]: Is this a todo ? :)

File: soda/core/soda/soda_cloud/soda_cloud.py [hunk 3]
```diff
@@ -179,6 +179,10 @@ def get_historic_data(self, historic_descriptor: HistoricDescriptor):
 
         return {"measurements": measurements, "check_results": check_results}
 
+    def get_check_attributes_schema(self) -> dict:
+        # TODO: implement.
```
Comment (vijaykiran) [inline soda/core/soda/soda_cloud/soda_cloud.py:183]: I think the attributes schema should be returned by the orgsettings call (which also contains failed rows enable/disable setting)

File: soda/core/soda/execution/check/check.py [hunk 4]
```diff
@@ -110,6 +110,12 @@ def __init__(
         self.partition: Partition = partition
         self.column: Column = column
         self.metrics: dict[str, Metric] = {}
+        self.attributes = {}
+        self.soda_attributes = (
```
Comment (vijaykiran) [inline soda/core/soda/execution/check/check.py:114]: We don't need soda_attributes  - everything is part of `atributes` - perhaps I wasnt' clear in the feature spec.

**Review by vijaykiran — CHANGES_REQUESTED** [review]
Looks pretty good! Please check the comments.

File: soda/core/soda/common/attributes_handler.py (hunk 1 above)
Comment (m1n0) [reply soda/core/soda/common/attributes_handler.py:13]: uh yes, fixed.

File: soda/core/soda/common/attributes_handler.py (hunk 2 above)
Comment (m1n0) [reply soda/core/soda/common/attributes_handler.py:61]: indeed, changed to except only incorrect type and value errors

File: soda/core/soda/execution/check/check.py (hunk 4 above)
Comment (m1n0) [reply soda/core/soda/execution/check/check.py:114]: removed, I was looking at outdated yaml spec 🤦

File: soda/core/soda/soda_cloud/soda_cloud.py (hunk 3 above)
Comment (m1n0) [reply soda/core/soda/soda_cloud/soda_cloud.py]: removed

**Review by vijaykiran — APPROVED** [review]

**Pull request merged**